File: cpp/a03.cpp
#include <iostream>
using namespace std;

int main() {
    int N, K;
    int P[109], Q[109];
    cin >> N >> K;
    for (int i = 0; i < N; i++)  cin >> P[i];
    for (int i = 0; i < N; i++)  cin >> Q[i];

    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            if ( P[i] + Q[j] == K ) {
                cout << "Yes";
                return 0;
            }
        }
    }
    cout << "No";
    return 0;
}

File: rust/a03.rs
use proconio::input;

fn main() {
    input! {
        n: u32,
        k: u32,
        p: [u32; n],
        q: [u32; n],
    }
    let mut is_found: bool = false;
    for i in p.iter() {
        for j in q.iter() {
            if i + j == k {
                is_found = true;
            }
        }
    }
    if is_found {
        println!("Yes")
    } else {
        println!("No")
    }
}
